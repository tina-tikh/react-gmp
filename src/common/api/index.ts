import * as MovieService from './movie-service';

export { MovieService };
