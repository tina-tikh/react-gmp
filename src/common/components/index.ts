import ErrorBoundary from './ErrorBoundary';
import FilmList from './FilmList';

export * from './form';
export * from './layout';
export { ErrorBoundary, FilmList };
