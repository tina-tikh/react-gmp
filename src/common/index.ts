export * from './api';
export * from './components';
export * from './models';
export * from './theme';
