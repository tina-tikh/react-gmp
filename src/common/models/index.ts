import Movie from './movie';

export { Movie };
