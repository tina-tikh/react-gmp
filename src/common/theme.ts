export type Theme = {
    colors: {
        primary: string,
        secondary: string,
        light: string,
        dark: string,
        darker: string,
        darkest: string
    }
};

const theme: Theme = {
    colors: {
        primary: '#f65261',
        secondary: '#a1e66f',
        light: '#fff',
        dark: '#555',
        darker: '#424242',
        darkest: '#232323'
    },
};

export default theme;
