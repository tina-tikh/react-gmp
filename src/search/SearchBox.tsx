import * as React from 'react';
import { PureComponent, ReactNode } from 'react';

import { Button, styled } from '../common';

type SearchBoxProps = {
  onSubmit: (query: string) => void;
};

type SearchBoxState = {
  searchValue: string;
};

const Container = styled.div`
  display: flex;
  justify-content: left;
  margin-bottom: 2rem;
`;

const Input = styled.input`
  flex: 1;
  max-width: 80rem;
  padding: 1.5rem;
  margin-right: 1rem;
  border: none;
  border-radius: 0.4rem;
  font-size: 2rem;
  color: ${(props) => props.theme.colors.light};
  background-color: rgba(255, 255, 255, 0.2);

  &:focus {
    outline: none;
  }
`;

const SearchButton = styled(Button)`
  width: 23rem;
  font-size: 2rem;
  font-weight: bold;
`;

class SearchBox extends PureComponent<SearchBoxProps, SearchBoxState> {
  constructor(props: SearchBoxProps) {
    super(props);
    this.state = {
      searchValue: '',
    };
  }

  render(): ReactNode {
    return (
      <Container>
        <Input
          placeholder="Search"
          value={this.state.searchValue}
          onChange={(e) => this.handleSearchValueChange(e)}
        ></Input>
        <SearchButton onClick={() => this.handleOnSubmit()}>
          Search
        </SearchButton>
      </Container>
    );
  }

  handleSearchValueChange(e: React.ChangeEvent<HTMLInputElement>): void {
    this.setState({
      searchValue: e.target.value,
    });
  }

  handleOnSubmit(): void {
    this.props.onSubmit(this.state.searchValue);
  }
}

export default SearchBox;
