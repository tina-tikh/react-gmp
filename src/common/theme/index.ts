import theme from './theme';
import styled from './styled';

export { styled, theme };
