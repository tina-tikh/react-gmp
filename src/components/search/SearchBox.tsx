import * as React from 'react';
import { PureComponent, ReactNode, RefObject } from 'react';

import { styled } from '../../theme';
import { Button } from '../form';

type SearchBoxProps = {
  searchValue: string;
  onSubmit?: (query: string) => void;
};

const Container = styled.div`
  display: flex;
  justify-content: left;
  margin-bottom: 2rem;
`;

const Input = styled.input`
  flex: 1;
  max-width: 80rem;
  padding: 1.5rem;
  margin-right: 1rem;
  border: none;
  border-radius: 0.4rem;
  font-size: 2rem;
  color: ${(props) => props.theme.colors.light};
  background-color: rgba(255, 255, 255, 0.2);

  &:focus {
    outline: none;
  }
`;

const SearchButton = styled(Button)`
  width: 23rem;
  font-size: 2rem;
  font-weight: bold;
`;

class SearchBox extends PureComponent<SearchBoxProps> {
  private inputRef: RefObject<HTMLInputElement> = React.createRef();

  handleKeyDown(event: React.KeyboardEvent<HTMLDivElement>): void {
    if (event.key === 'Enter') {
      event.preventDefault();
      event.stopPropagation();
      this.handleOnSubmit();
    }
  }

  handleOnSubmit(): void {
    const { current: { value } } = this.inputRef;
    const { onSubmit } = this.props;

    onSubmit(value);
  }

  render(): ReactNode {
    const { searchValue } = this.props;
    return (
      <Container>
        <Input
          ref={this.inputRef}
          placeholder="Search"
          defaultValue={searchValue}
          onKeyDown={(e) => this.handleKeyDown(e)}
        />
        <SearchButton onClick={() => this.handleOnSubmit()}>
          Search
        </SearchButton>
      </Container>
    );
  }
}

export default SearchBox;
