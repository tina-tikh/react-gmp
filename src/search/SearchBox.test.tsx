import * as React from 'react';
import { mount, shallow } from 'enzyme';

import { Page } from '../common';
import SearchBox from './SearchBox';

describe('<SearchBox />', () => {
  it('should be rendered', () => {
    const component = shallow(<SearchBox />);
    expect(component).toMatchSnapshot();
  });

  it('should control input', () => {
    const inputValue = '12345';
    const component = mount(
      <Page>
        <SearchBox />
      </Page>
    );
    component
      .find('input')
      .simulate('change', { target: { value: inputValue } });
    expect(component.find(SearchBox).state().searchValue).toEqual(inputValue);
  });

  it('should call onSubmit', () => {
    const onSubmit = jest.fn();
    const component = mount(
      <Page>
        <SearchBox onSubmit={onSubmit} />
      </Page>
    );
    component.find('button').simulate('click');
    expect(onSubmit).toHaveBeenCalled();
  });
});
