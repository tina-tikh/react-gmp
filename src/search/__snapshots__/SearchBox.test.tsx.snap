// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> should be rendered 1`] = `
<Styled(div)>
  <Styled(input)
    onChange={[Function]}
    placeholder="Search"
    value=""
  />
  <Styled(button)
    onClick={[Function]}
  >
    Search
  </Styled(button)>
</Styled(div)>
`;
